### PR TITLE
azure: Populate node labels from tags

### PR DESCRIPTION
**Pull request by hakman**
/cc @justinsb @rifelpet

### CODE REVIEW

File: pkg/nodeidentity/azure/identify.go [hunk 1]
```diff
@@ -100,24 +103,41 @@ func (i *nodeIdentifier) IdentifyNode(ctx context.Context, node *corev1.Node) (*
...
+		if k == azure.TagClusterName && v != nil {
+			labels[kops.LabelClusterName] = *v
+		}
+		if k == InstanceGroupNameTag && v != nil {
+			labels[kops.NodeLabelInstanceGroup] = *v
+		}
+		if strings.HasPrefix(k, azure.TagNameRolePrefix) {
+			role := strings.TrimPrefix(k, azure.TagNameRolePrefix)
+			switch {
+			case strings.HasSuffix(nodelabels.RoleLabelControlPlane20, role):
```
Comment (justinsb) [inline pkg/nodeidentity/azure/identify.go:116]: Should this just be checking against the instancegroup roles (and I think you've already stripped the prefix)?

https://github.com/kubernetes/kops/blob/d5c24585187ddb50647fa939cb4e7fa8d997b077/pkg/model/azuremodel/context.go#L117C1-L121

**Comment by justinsb** [comment]
Thanks @hakman - and thanks for tweaking the code I didn't grok!

/approve
/lgtm

File: pkg/nodeidentity/azure/identify.go (hunk 1 above)
Comment (hakman) [reply pkg/nodeidentity/azure/identify.go]: Azure tag is in the format `"k8s.io_role_control-plane": "1"`, so we have to trim the `k8s.io_role_` prefix.
I agree that it looks better now with the `InstanceGroupRoleX.ToLowerString()`.

**Pull request merged**